### PR TITLE
Add pre-commencement conditions to review

### PR DESCRIPTION
**Pull request by benjamineskola**
### Description of change

Allow reviewing pre-commencement conditions.

### Story Link

https://trello.com/c/8b0c3dOQ/2665-reviewer-is-able-to-see-that-pre-commencement-conditions-have-been-agreed

### Screenshots

Screenshot 2024-03-26 at 14 57 41

### Decisions [OPTIONAL]

If you had to make any decisions between different ways of doing things, what where they and why?

### Known issues [OPTIONAL]

Things you know need further follow on work but aren't in scope of this issue

1. issue1
2. issue2

### Further testing or sign off required [OPTIONAL]

e.g.

1. product manager to sign off view templates

### CODE REVIEW

File: app/views/planning_applications/review/tasks/_review_assessment.html.erb [hunk 1]
```diff
@@ -44,12 +44,20 @@
     )%>
   <% end %>
   <% if @planning_application.application_type.planning_conditions? %>
-    <% if @planning_application.condition_set.present? && @planning_application.condition_set.current_review&.status != "not_started" %>
+    <% if @planning_application.condition_set&.current_review&.started? %>
```
Comment (cdccollins) [inline app/views/planning_applications/review/tasks/_review_assessment.html.erb:47]: Just double checking - this link will still show if there aren't any conditions?

File: spec/system/planning_applications/review/pre_commencement_conditions_spec.rb [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+          "Review pre-commencement conditions",
+          with: "To be reviewed"
+        )
+
+        condition_set = planning_application.pre_commencement_condition_set
+        expect(condition_set.current_review.action).to eq "rejected"
+        expect(condition_set.current_review.comment).to eq "I don't think you've assessed conditions correctly"
+        expect(condition_set.current_review.status).to eq "to_be_reviewed"
+
+        # sign_out(reviewer)
```
Comment (cdccollins) [inline spec/system/planning_applications/review/pre_commencement_conditions_spec.rb:119]: Should this all be commented out?

**Review by cdccollins — APPROVED** [review]

File: app/views/planning_applications/review/tasks/_review_assessment.html.erb (hunk 1 above)
Comment (benjamineskola) [reply app/views/planning_applications/review/tasks/_review_assessment.html.erb:47]: I think it ought to be the same logic, just simplified by turning `status != "not_started"` into a method, and using `&.` to make the `present?` check unnecessary.

It'll not show if the condition set is nil, which I think is the same as before.

File: spec/system/planning_applications/review/pre_commencement_conditions_spec.rb (hunk 2 above)
Comment (benjamineskola) [reply spec/system/planning_applications/review/pre_commencement_conditions_spec.rb:119]: No. 🤦🏻 I've just uncommented and realised that the reviewer comment doesn't show when returned to the assessor.

File: app/views/planning_applications/review/pre_commencement_conditions/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  <div class="govuk-grid-column-full">
+    <h2 class="govuk-heading-m">Summary of pre-commencement conditions</h2>
+    <% if @planning_application.pre_commencement_condition_set&.approved_conditions.present? %>
+      <ol class="govuk-list govuk-list--number">
+        <% @planning_application.pre_commencement_condition_set.approved_conditions.sorted.each do |condition| %>
+          <li>
+            <p class="govuk-body govuk-!-margin-bottom-2">
+              <% if condition.title? %>
+                <span class="govuk-!-font-weight-bold"><%= condition.title %></span><br>
+                <%= condition.text %><br>
```
[the remark this answers is not in the document]
Comment (cdccollins) [reply app/views/planning_applications/review/pre_commencement_conditions/edit.html.erb:27]: (applies to the other places where you show the text/reason

File: app/controllers/planning_applications/review/pre_commencement_conditions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  class PreCommencementConditionsController < AuthenticationController
+    include CommitMatchable
+    include PlanningApplicationAssessable
+
+    before_action :set_planning_application
+    before_action :ensure_planning_application_is_validated
+    before_action :ensure_user_is_reviewer
+
+    def show
+    end
```
Comment (benbaumann95) [inline app/controllers/planning_applications/review/pre_commencement_conditions_controller.rb:13]: For consistency should we add?
```ruby
respond_to do |format|
  format.html
end
```

**Review by benbaumann95 — APPROVED** [review]

File: app/controllers/planning_applications/review/pre_commencement_conditions_controller.rb (hunk 4 above)
Comment (benjamineskola) [reply app/controllers/planning_applications/review/pre_commencement_conditions_controller.rb:13]: I wasn't sure how necessary this was, but I think it's better to be explicit, yes.

**Pull request merged**